argcomplete/__init__.py

```diff
@@ -321,16 +332,19 @@ def __call__(self, parser, namespace, values, option_string=None):
         return self.active_parsers
 
     def _get_subparser_completions(self, parser, cword_prefix):
-        def filter_aliases(metavar, dest, prefix):
-            if not metavar:
-                return dest if dest and dest.startswith(prefix) else ""
-
-            # metavar combines dest and aliases with ",".
-            a = metavar.replace(",", "").split()
-            return " ".join(x for x in a if x.startswith(prefix))
+        def filter_aliases(aliases, prefix):
+            return tuple(x for x in aliases if x.startswith(prefix))
+
+        aliases_by_parser = {}
+        for key in parser.choices.keys():
+            p = parser.choices[key]
+            if not aliases_by_parser.get(p):
```

Comment:
Probably should change this to `defaultdict`

Reply:
Agree, or you could use `aliases_by_parser.setdefault(p, []).append(key)`

Reply:
changed to `setdefault`